src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DocumentPresentation/TextDocumentUriPresentationEndpoint.cs

```diff
@@ -81,24 +81,26 @@ protected override IRazorPresentationParams CreateRazorRequestParameters(UriPres
             return null;
         }
 
+        var razorFileUri = request.Uris.Where(
+            x => Path.GetFileName(x.GetAbsoluteOrUNCPath()).EndsWith(".razor", FilePathComparison.Instance)).LastOrDefault();
```

Comment:
Should use `FirstOrDefault` here instead of `LastOrDefault`, since we don't care, and its strictly speaking a teensy bit faster, since it can stop at the first match.